Fix the 'Build, Sign and Publish' step failure

carlossanlop opened the pull request:
Fixes https://github.com/dotnet/maintenance-packages/issues/27

Explanation in the issue.

carlossanlop commented:
@ericstj @ViktorHofer I do not know if it worked, since the problem shows up in the rolling builds, not in PRs. Do we have azp run help here?

carlossanlop commented:
/azp run list

ericstj commented:
You'd need to submit an internal build from a branch off the internal repo.  Public github PRs cannot trigger internal signed builds.

I double checked the binlog for the build here, and this property is already set.
https://artprodcus3.artifacts.visualstudio.com/Ab55de4ed-4b5a-4215-a8e4-0a0a5f71e7d8/7ea9116e-9fac-403d-b258-b31fcf1bb293/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RuY2VuZy9wcm9qZWN0SWQvN2VhOTExNmUtOWZhYy00MDNkLWIyNTgtYjMxZmNmMWJiMjkzL2J1aWxkSWQvMjMyOTU5OC9hcnRpZmFjdE5hbWUvTG9nc19CdWlsZF9XaW5kb3dzX05UX1JlbGVhc2U1/content?format=file&subPath=%2FRelease%2FBuild.binlog
<img width="382" alt="image" src="https://github.com/dotnet/maintenance-packages/assets/8918108/913f2109-9668-4c2b-bc61-49f1b9a7db7c">

That property seems to be set as an environment variable on the build machine which might explain why you can't repro locally.

ericstj commented:
I think I found the change you need to make, comments are in the original issue. https://github.com/dotnet/maintenance-packages/issues/27#issuecomment-1841721518

carlossanlop commented:
Alright, I pushed the suggested change and reverted the original. I renamed the PR, and also pushed the branch to internal so I can test [the pipeline](https://dev.azure.com/dnceng/internal/_build/results?buildId=2330129&view=results).

carlossanlop commented:
My pipeline trigger didn't reach the signing step. It failed in the nuget validation step: https://dev.azure.com/dnceng/internal/_build/results?buildId=2330129&view=logs&j=a13e9384-d6d3-5a14-7a96-79282fc11262&t=40bbd2a4-8718-563d-213e-e39022947e46

I guess that's expected since no package was enabled.
